### PR TITLE
Add content_format option for File.Source

### PR DESCRIPTION
**Pull request by Rados13**
Changes in PR:

- add option `content_format` for `File.Source` 
- update deps
- update credo

### CODE REVIEW

File: lib/membrane_file/source.ex [hunk 1]
```diff
@@ -17,17 +17,27 @@ defmodule Membrane.File.Source do
...
 
   def_output_pad :output, caps: {RemoteStream, type: :bytestream}
 
   @impl true
-  def handle_init(%__MODULE__{location: location, chunk_size: size}) do
+  def handle_init(%__MODULE__{
+        location: location,
+        chunk_size: size,
+        content_format: content_format
+      }) do
```
Comment (bblaszkow06) [inline lib/membrane_file/source.ex:34]: This started to grow too much. I guess you can just do

```suggestion
  def handle_init(options) do
    state =
      options
      |> Map.from_struct()
      |> Map.update(:location, &Path.expand/1)
      
    {:ok, state}
  end
```

File: lib/membrane_file/source.ex [hunk 2]
```diff
@@ -17,17 +17,27 @@ defmodule Membrane.File.Source do
                 spec: pos_integer(),
                 default: 2048,
                 description: "Size of chunks being read"
+              ],
+              content_format: [
+                spec: module() | nil,
+                default: nil,
+                description: "Caps which will be sent from Source"
```
Comment (bblaszkow06) [inline lib/membrane_file/source.ex:24]: This description seems to be taken from the old version of the PR :wink: 
I'd rephrase it to better reflect what will happen with that content_format

File: lib/membrane_file/source.ex [hunk 3]
```diff
@@ -39,7 +49,9 @@ defmodule Membrane.File.Source do
 
   @impl true
   def handle_prepared_to_playing(_ctx, state) do
-    {{:ok, caps: {:output, %RemoteStream{type: :bytestream}}}, state}
+    {{:ok,
+      caps: {:output, %RemoteStream{type: :bytestream, content_format: state.content_format}}},
+     state}
```
Comment (bblaszkow06) [inline lib/membrane_file/source.ex:54]:
```suggestion
    output_format = %RemoteStream{type: :bytestream, content_format: state.content_format}
    {{:ok, caps: {:output, output_format}}, state}
```

**Review by bblaszkow06 — CHANGES_REQUESTED** [review]

File: lib/membrane_file/source.ex [hunk 4]
```diff
@@ -17,18 +17,24 @@ defmodule Membrane.File.Source do
                 spec: pos_integer(),
                 default: 2048,
                 description: "Size of chunks being read"
+              ],
+              content_format: [
+                spec: module() | nil,
+                default: nil,
+                description: "Module for `RemoteStream.t()` caps sent by output pad"
               ]
```
Comment (mat-hek) [inline lib/membrane_file/source.ex:25]: Still don't think this is the way. Whatever accepts remote stream should accept `nil` as the content format.

**Review by mat-hek — CHANGES_REQUESTED** [review]

**Comment by Rados13** [comment]
After talking with @mat-hek, these changes don't make sense.
